ast/ddl.go

```diff
@@ -1790,23 +1791,24 @@ type AlterTableSpec struct {
 	// see https://mariadb.com/kb/en/library/alter-table/
 	IfNotExists bool
 
-	Tp              AlterTableType
-	Name            string
-	Constraint      *Constraint
-	Options         []*TableOption
-	NewTable        *TableName
-	NewColumns      []*ColumnDef
-	OldColumnName   *ColumnName
-	Position        *ColumnPosition
-	LockType        LockType
-	Algorithm       AlterAlgorithm
-	Comment         string
-	FromKey         model.CIStr
-	ToKey           model.CIStr
-	Partition       *PartitionOptions
-	PartitionNames  []model.CIStr
-	PartDefinitions []*PartitionDefinition
-	Num             uint64
+	Tp               AlterTableType
+	Name             string
+	Constraint       *Constraint
+	Options          []*TableOption
+	NewTable         *TableName
+	NewColumns       []*ColumnDef
+	OldColumnName    *ColumnName
+	Position         *ColumnPosition
+	LockType         LockType
+	Algorithm        AlterAlgorithm
+	Comment          string
+	FromKey          model.CIStr
+	ToKey            model.CIStr
+	Partition        *PartitionOptions
+	PartitionNames   []model.CIStr
+	PartDefinitions  []*PartitionDefinition
+	Num              uint64
+	IfWithValidation bool
```

Comment:
Adding a type is better than adding a member because we should make the changes to the data structure as small as possible.